contents/docs/cdp/destinations/advanced.md

```diff
@@ -0,0 +1,133 @@
+
+You can template destination output using curly braces `{}`.
+
+For example, given this PostHog event:
+
+```ts
+{
+    event: {
+        event: "$pageview",
+        distinct_id: "1234",
+        timestamp: "2024-01-01T12:00:00",
+        properties: {
+            $current_url: "https://posthog.com"
+        }
+    },
+    person: {
+        name: "max@posthog.com",
+        properties: {
+            email: "max@posthog.com",
+            first_name: "Max",
+            last_name: "the Hedgehog"
+        }
+    }
+}
+```
+
+You can create a template like:
+
+```
+{event.event} was triggered by {person.properties.first_name} {person.properties.last_name}
+
+// Outputs: $pageview was triggered by Max the Hedgehog
+```
+
+You could use this approach to set parameters in a [webhook](/docs/cdp/destinations/webhook) destination’s URL:
+
+```
+https://api.yourgreatwebsite/v0/track-conversion?username={person.name}
+```
+
+Or to construct a custom JSON payload that conforms to an existing API specification:
+
+```ts
+{
+    "event_source": "posthog",
+    "event": {
+        "event_label": "conversion",
+        "username": "{person.name}",
+        "datetime": "{event.timestamp}"
+    }
+}
+```
+
+### Global object
+
+Below is the structure of the global variables available whenever templating a destination.
+
+```ts
+{
+    event: {
+        uuid: string // The unique ID of the event
+        event: string // The event name (e.g. $pageview)
+        distinct_id: string // The distinct_id of the identity that created the event
+        properties: Record<string, any>
+        timestamp: string
+        url: string // A URL to view it in PostHog
+    }
+    person?: {
+        id: string // The UUID of the Person associated with the distinct_id of the event
+        name: string // Configured based on your "Display name" property in PostHog
+        url: string  // A URL to view it in PostHog
+        properties: Record<string, any>
+    }
+    groups?: {
+        [id: string]: {
+            id: string
+            type: string
+            index: number
+            url: string  // A URL to view it in PostHog
+            properties: Record<string, any>
+        }
+    }
+    project: {
+        id: number  // The ID of the PostHog project
+        name: string  // The name of the PostHog project
+        url: string  // A URL to view it in PostHog
+    }
+    source?: {
+        name: string // The name of the source (typically the destination name)
+        url: string  // A URL to view it in PostHog
+    }
+}
+```
+
+## Going deeper with Hog
+
+For most cases, we recommend using one of the templates. These take care of most logic under the hood, exposing simple inputs for you to configure. 
+
+You can, however, modify any destination by clicking `show source code`. From here you can modify the inputs – for example marking an input as secret so that it is encrypted as rest – or change the implementation of the destination itself.
```

Comment:
```suggestion
You can, however, modify any destination by clicking `show source code`. From here you can modify the inputs – for example marking an input as secret so that it is encrypted as rest – or changing the implementation of the destination itself.
```